pkg/serializer/frame/utils.go

```diff
@@ -0,0 +1,96 @@
+package frame
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing/iotest"
+)
+
+// FrameList is a list of frames (byte arrays), used for convenience functions
+type FrameList [][]byte
+
+// ReadFrameList is a convenience method that reads all available frames from the Reader
+// into a returned FrameList. If an error occurs, reading stops and the error is returned.
+func ReadFrameList(ctx context.Context, fr Reader) (FrameList, error) {
+	var frameList FrameList
+	for {
+		// Read until we get io.EOF or an error
+		frame, err := fr.ReadFrame(ctx)
+		if errors.Is(err, io.EOF) {
+			break
+		} else if err != nil {
```

Comment:
You don't need the else here, you've already executed the `break` if it was an `io.EOF` error?

https://gist.github.com/adamveld12/c0d9f0d5f0e1fba1e551#indent-error-flow